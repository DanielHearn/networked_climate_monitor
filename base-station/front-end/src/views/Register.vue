<template>
  <main-panel>
    <template slot="content">
      <div class="content-column">
        <h1 class="title">Climate Monitor</h1>
        <h2 class="heading">Account Registration</h2>
        <p class="text">
          Enter an email and password to register the account.
        </p>
        <form id="register" class="form" @submit="checkRegister" method="post">
          <div class="input-box">
            <p class="text">Email</p>
            <input
              type="email"
              class="input--text"
              name="email"
              v-model="email"
              placeholder="Email"
              tabindex="0"
            />
          </div>
          <div class="input-box">
            <p class="text">Password</p>
            <input
              type="password"
              class="input--text"
              name="password"
              v-model="password"
              placeholder="Password"
              tabindex="0"
            />
          </div>
          <div class="input-box">
            <p class="text">Confirm Password</p>
            <input
              type="password"
              class="input--text"
              name="confirmPassword"
              v-model="confirmPassword"
              placeholder="Confirm password"
              tabindex="0"
            />
          </div>
          <error-list
            v-if="errors.length"
            title="Errors:"
            v-bind:errors="errors"
          />
          <v-button
            :hierachyLevel="'primary'"
            :text="'Register'"
            type="submit"
            tabindex="0"
          />
        </form>
        <router-link to="/login" class="link"
          >Already registered? Click here to login</router-link
        >
      </div>
    </template>
  </main-panel>
</template>

<script>
import ErrorList from './../components/ErrorList/ErrorList.vue'
import MainPanel from './../components/MainPanel/MainPanel.vue'
import { HTTP } from './../static/http-common'
import { processErrors } from './../static/helpers'
import {
  setStoredAccessToken,
  setStoredRefreshToken
} from './../store/storage.js'
import vButton from './../components/vButton/vButton.vue'

export default {
  name: 'register',
  components: {
    MainPanel,
    ErrorList,
    vButton
  },
  data: function() {
    return {
      email: '',
      password: '',
      confirmPassword: '',
      errors: []
    }
  },
  methods: {
    checkRegister: function(e) {
      e.preventDefault()
      this.errors = []

      const email = this.email
      const password = this.password
      const newPassword = this.confirmPassword
      let valid = true

      if (email === '') {
        this.errors.push('Enter a email.')
        valid = false
      } else if (email.indexOf('@') === -1) {
        this.errors.push('Enter a valid email.')
        valid = false
      }
      if (password === '') {
        this.errors.push('Enter a password.')
        valid = false
      } else if (password.length < 8 || password.length > 40) {
        this.errors.push('Password must be between 8 and 40 characters long.')
        valid = false
      }

      if (password !== newPassword) {
        this.errors.push('Password and confirm password be identical.')
        valid = false
      }

      if (valid) {
        this.register(email, password)
      }
    },
    register: function(email, password) {
      HTTP.post('account', {
        email: email,
        password: password
      })
        .then(response => {
          const data = response.data
          if (data.status && data.access_token && data.refresh_token) {
            const user = this.$store.state.user
            user.logged_in = true
            user.access_token = data.access_token
            user.refresh_token = data.refresh_token
            user.email = email

            setStoredAccessToken(data.access_token)
            setStoredRefreshToken(data.refresh_token)

            HTTP.get('account', {
              headers: { Authorization: 'Bearer ' + data.access_token }
            })
              .then(response => {
                const data = response.data
                if (data.status && data.account) {
                  const user = this.$store.state.user
                  user.settings = JSON.parse(
                    data.account.settings.replace(/'/g, '"')
                  )

                  this.$store.commit('setUser', user)
                }
              })
              .catch(e => {
                console.log(e)
              })

            this.$store.commit('setUser', user)
            this.$toasted.show('Registered')
            this.$toasted.show('Logged in')

            this.$router.push('/dashboard')
          }
        })
        .catch(e => {
          if (e.response && e.response.data && e.response.data.errors) {
            this.errors = processErrors(e.response.data.errors)
          } else {
            this.errors.push(e)
          }
        })
    }
  }
}
</script>
