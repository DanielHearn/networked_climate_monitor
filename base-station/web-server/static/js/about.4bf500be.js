(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["about"],{be4d:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",[n("p",[t._v("404 - Not Found")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports},f820:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"about"},[n("h1",[t._v("This is an about page")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports}}]);
//# sourceMappingURL=about.4bf500be.js.map