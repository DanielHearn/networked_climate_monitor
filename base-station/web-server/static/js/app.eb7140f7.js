(function(t){function e(e){for(var r,o,i=e[0],c=e[1],u=e[2],l=0,d=[];l<i.length;l++)o=i[l],Object.prototype.hasOwnProperty.call(s,o)&&s[o]&&d.push(s[o][0]),s[o]=0;for(r in c)Object.prototype.hasOwnProperty.call(c,r)&&(t[r]=c[r]);f&&f(e);while(d.length)d.shift()();return a.push.apply(a,u||[]),n()}function n(){for(var t,e=0;e<a.length;e++){for(var n=a[e],r=!0,o=1;o<n.length;o++){var i=n[o];0!==s[i]&&(r=!1)}r&&(a.splice(e--,1),t=c(c.s=n[0]))}return t}var r={},o={app:0},s={app:0},a=[];function i(t){return c.p+"static/js/"+({"dashboard~login~register~resetpassword":"dashboard~login~register~resetpassword",dashboard:"dashboard",login:"login",register:"register",resetpassword:"resetpassword",notfound:"notfound",settings:"settings"}[t]||t)+"."+{"dashboard~login~register~resetpassword":"db62f035",dashboard:"7b353a62",login:"6dc38a8a",register:"f4245d36",resetpassword:"e339c592",notfound:"0072178d",settings:"dbfbf29e"}[t]+".js"}function c(e){if(r[e])return r[e].exports;var n=r[e]={i:e,l:!1,exports:{}};return t[e].call(n.exports,n,n.exports,c),n.l=!0,n.exports}c.e=function(t){var e=[],n={dashboard:1,login:1,register:1,resetpassword:1,settings:1};o[t]?e.push(o[t]):0!==o[t]&&n[t]&&e.push(o[t]=new Promise((function(e,n){for(var r="static/css/"+({"dashboard~login~register~resetpassword":"dashboard~login~register~resetpassword",dashboard:"dashboard",login:"login",register:"register",resetpassword:"resetpassword",notfound:"notfound",settings:"settings"}[t]||t)+"."+{"dashboard~login~register~resetpassword":"31d6cfe0",dashboard:"72b473cd",login:"5ae2af0a",register:"5ae2af0a",resetpassword:"5ae2af0a",notfound:"31d6cfe0",settings:"e5f34ba8"}[t]+".css",s=c.p+r,a=document.getElementsByTagName("link"),i=0;i<a.length;i++){var u=a[i],l=u.getAttribute("data-href")||u.getAttribute("href");if("stylesheet"===u.rel&&(l===r||l===s))return e()}var d=document.getElementsByTagName("style");for(i=0;i<d.length;i++){u=d[i],l=u.getAttribute("data-href");if(l===r||l===s)return e()}var f=document.createElement("link");f.rel="stylesheet",f.type="text/css",f.onload=e,f.onerror=function(e){var r=e&&e.target&&e.target.src||s,a=new Error("Loading CSS chunk "+t+" failed.\n("+r+")");a.code="CSS_CHUNK_LOAD_FAILED",a.request=r,delete o[t],f.parentNode.removeChild(f),n(a)},f.href=s;var h=document.getElementsByTagName("head")[0];h.appendChild(f)})).then((function(){o[t]=0})));var r=s[t];if(0!==r)if(r)e.push(r[2]);else{var a=new Promise((function(e,n){r=s[t]=[e,n]}));e.push(r[2]=a);var u,l=document.createElement("script");l.charset="utf-8",l.timeout=120,c.nc&&l.setAttribute("nonce",c.nc),l.src=i(t);var d=new Error;u=function(e){l.onerror=l.onload=null,clearTimeout(f);var n=s[t];if(0!==n){if(n){var r=e&&("load"===e.type?"missing":e.type),o=e&&e.target&&e.target.src;d.message="Loading chunk "+t+" failed.\n("+r+": "+o+")",d.name="ChunkLoadError",d.type=r,d.request=o,n[1](d)}s[t]=void 0}};var f=setTimeout((function(){u({type:"timeout",target:l})}),12e4);l.onerror=l.onload=u,document.head.appendChild(l)}return Promise.all(e)},c.m=t,c.c=r,c.d=function(t,e,n){c.o(t,e)||Object.defineProperty(t,e,{enumerable:!0,get:n})},c.r=function(t){"undefined"!==typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},c.t=function(t,e){if(1&e&&(t=c(t)),8&e)return t;if(4&e&&"object"===typeof t&&t&&t.__esModule)return t;var n=Object.create(null);if(c.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:t}),2&e&&"string"!=typeof t)for(var r in t)c.d(n,r,function(e){return t[e]}.bind(null,r));return n},c.n=function(t){var e=t&&t.__esModule?function(){return t["default"]}:function(){return t};return c.d(e,"a",e),e},c.o=function(t,e){return Object.prototype.hasOwnProperty.call(t,e)},c.p="/",c.oe=function(t){throw console.error(t),t};var u=window["webpackJsonp"]=window["webpackJsonp"]||[],l=u.push.bind(u);u.push=e,u=u.slice();for(var d=0;d<u.length;d++)e(u[d]);var f=l;a.push([0,"chunk-vendors"]),n()})({0:function(t,e,n){t.exports=n("56d7")},"0513":function(t,e,n){},"13b9":function(t,e,n){},1695:function(t,e,n){"use strict";var r=n("b1fe"),o=n.n(r);o.a},"2eaa":function(t,e,n){},4733:function(t,e,n){"use strict";var r=n("2eaa"),o=n.n(r);o.a},"4ccb":function(t,e,n){"use strict";var r=n("0513"),o=n.n(r);o.a},"56d7":function(t,e,n){"use strict";n.r(e);n("e260"),n("e6cf"),n("cca6"),n("a79d");var r=n("2b0e"),o=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{class:{"mobile-menu--active":t.$store.state.mobileMenu,mobile:t.$store.state.mobile},attrs:{id:"app"}},[n("v-nav"),t.$store.state.mobileMenu?n("mobile-menu"):t._e(),n("router-view",{class:{hidden:t.$store.state.mobileMenu}})],1)},s=[],a=(n("ac1f"),n("5319"),n("ecf6")),i=n("66ca"),c=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"nav"},[n("div",{staticClass:"nav__side nav__left"},[t.$store.state.mobile?n("logout-button"):n("router-link",{staticClass:"link title--nav",attrs:{to:"/"}},[t._v("Climate Monitor")])],1),n("div",{staticClass:"nav__links"},[t.$store.state.mobile?t._e():[n("router-link",{staticClass:"link",attrs:{to:"/"}},[t._v("Home")]),t.$store.state.user.logged_in?t._e():n("router-link",{staticClass:"link",attrs:{to:"/login"}},[t._v("Login")]),t.$store.state.user.logged_in?t._e():n("router-link",{staticClass:"link",attrs:{to:"/register"}},[t._v("Register")]),t.$store.state.user.logged_in?n("router-link",{staticClass:"link",attrs:{to:"/dashboard"}},[t._v("Dashboard")]):t._e(),t.$store.state.user.logged_in?n("router-link",{staticClass:"link",attrs:{to:"/settings"}},[t._v("Settings")]):t._e()]],2),n("div",{staticClass:"nav__side nav__right"},[t.$store.state.user.logged_in&&!t.$store.state.mobile?n("logout-button"):t._e(),t.$store.state.mobile?n("v-button",{staticStyle:{"margin-left":"2em"},attrs:{hierachyLevel:"secondary",text:t.$store.state.mobileMenu?"close":"menu",isIcon:!0},nativeOn:{click:function(e){return t.toggleMenu(e)}}}):t._e()],1)])},u=[],l=n("c860"),d=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("v-button",{attrs:{hierachyLevel:"secondary",text:"person",isIcon:!0},nativeOn:{click:function(e){return t.logout(e)}}})},f=[],h=(n("b0c0"),{name:"logout-button",components:{vButton:l["a"]},methods:{logout:function(){var t=this.$store.state.user,e=t.access_token,n=t.refresh_token;t.logged_in=!1,t.access_token="",t.refresh_token="",t.email="",this.$store.commit("setUser",t),Object(i["c"])(""),Object(i["d"])(""),this.$toasted.show("Logged out"),"home"!==this.$route.name&&this.$router.push("/"),Object(a["h"])(e).then((function(){})).catch((function(t){console.warn(t.response)})),Object(a["i"])(n).then((function(){})).catch((function(t){console.warn(t.response)}))}}}),g=h,m=(n("4ccb"),n("2877")),p=Object(m["a"])(g,d,f,!1,null,null,null),b=p.exports,v={name:"v-nav",components:{vButton:l["a"],LogoutButton:b},methods:{toggleMenu:function(){this.$store.commit("setMobileMenu",!this.$store.state.mobileMenu)}}},_=v,k=(n("1695"),Object(m["a"])(_,c,u,!1,null,null,null)),w=k.exports,y=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"mobile-menu"},[n("ul",{on:{click:t.toggleMenu}},[n("router-link",{staticClass:"link",attrs:{to:"/"}},[t._v("Home")]),t.$store.state.user.logged_in?t._e():[n("router-link",{staticClass:"link",attrs:{to:"/login"}},[t._v("Login")]),n("router-link",{staticClass:"link",attrs:{to:"/register"}},[t._v("Register")])],t.$store.state.user.logged_in?[t.$store.state.user.logged_in?n("router-link",{staticClass:"link",attrs:{to:"/dashboard"}},[t._v("Dashboard")]):t._e(),t.$store.state.user.logged_in?n("router-link",{staticClass:"link",attrs:{to:"/settings"}},[t._v("Settings")]):t._e()]:t._e()],2)])},$=[],C={name:"mobile-menu",methods:{toggleMenu:function(){this.$store.commit("setMobileMenu",!this.$store.state.mobileMenu)}}},O=C,j=(n("a3f9"),Object(m["a"])(O,y,$,!1,null,null,null)),S=j.exports,M={name:"app",components:{vNav:w,MobileMenu:S},methods:{checkScreenSize:function(){this.$store.commit("setMobile",window.innerWidth<=800)},loginFromStorage:function(){var t=this,e=Object(i["a"])(),n=Object(i["b"])();e&&Object(a["d"])(e).then((function(r){var o=r.data;if(o.status&&o.account){var s=Object.assign({},t.$store.state.user);s.logged_in=!0,s.access_token=e,s.refresh_token=n,s.email=o.account.email,s.reset_token=o.account.reset_token,s.settings=JSON.parse(o.account.settings.replace(/'/g,'"')),t.$store.commit("setUser",s),t.$toasted.show("Logged in")}})).catch((function(e){e.response?t.$toasted.show("Stored account credentials are invalid"):e.request&&t.$toasted.show("Error contacting server")}))}},created:function(){this.loginFromStorage(),this.checkScreenSize(),window.addEventListener("resize",this.checkScreenSize)}},P=M,x=(n("5c0b"),Object(m["a"])(P,o,s,!1,null,null,null)),E=x.exports,L=(n("d3b7"),n("8c4f")),A=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("main-panel",[n("template",{slot:"content"},[n("div",{staticClass:"content-column"},[n("h2",{staticClass:"title"},[t._v("Climate Monitor")]),n("h3",{staticClass:"heading"},[t._v("Home")]),n("p",[t._v(" The networked climate monitor project provides access to the recent and historical climate data from the sensors. ")]),n("p",[t._v(" This project has been designed and developed by Daniel Hearn for a final year project at the University of Portsmouth. The source code and designs for this project can be found on "),n("a",{staticClass:"link",attrs:{href:"https://github.com/DanielHearn/networked_climate_monitor"}},[t._v("Github")])]),n("p",[t._v(" To use the project the sensor nodes and base station must be setup, turned on and running their software. You must then register an account and login to this account. You can then access the climate data and manage sensors. ")]),n("p",[n("router-link",{staticClass:"link",attrs:{to:"/register"}},[t._v("To register click here")]),t._v(" or "),n("router-link",{staticClass:"link",attrs:{to:"/register"}},[t._v("click here to login")]),t._v(" or "),n("router-link",{staticClass:"link",attrs:{to:"/dashboard"}},[t._v("click here to view the dashboard.")])],1)])])],2)},B=[],T=n("af63"),z={name:"home",components:{MainPanel:T["a"]}},N=z,I=Object(m["a"])(N,A,B,!1,null,null,null),H=I.exports,D=n("4362"),J=n.n(D);r["default"].use(L["a"]);var U=" - Climate Monitor",q=[{path:"/",name:"home",component:H,meta:{title:function(){return"Home".concat(U)}}},{path:"/login",name:"login",component:function(){return Promise.all([n.e("dashboard~login~register~resetpassword"),n.e("login")]).then(n.bind(null,"a55b"))},meta:{title:function(){return"Login".concat(U)}}},{path:"/register",name:"register",component:function(){return Promise.all([n.e("dashboard~login~register~resetpassword"),n.e("register")]).then(n.bind(null,"73cf"))},meta:{title:function(){return"Register".concat(U)}}},{path:"/reset-password",name:"resetpassword",component:function(){return Promise.all([n.e("dashboard~login~register~resetpassword"),n.e("resetpassword")]).then(n.bind(null,"0813"))},meta:{title:function(){return"Reset Password".concat(U)}}},{path:"/dashboard",name:"dashboard",component:function(){return Promise.all([n.e("dashboard~login~register~resetpassword"),n.e("dashboard")]).then(n.bind(null,"7277"))},meta:{title:function(){return"Dashboard".concat(U)}}},{path:"/settings",name:"settings",component:function(){return n.e("settings").then(n.bind(null,"26d3"))},meta:{title:function(){return"Settings".concat(U)}}},{path:"*",component:function(){return n.e("notfound").then(n.bind(null,"9703"))},meta:{title:function(){return"404 Not Found".concat(U)}}}],R=new L["a"]({mode:"history",base:J.a.env.BASE_URL,routes:q}),F=R;R.afterEach((function(t){r["default"].nextTick((function(){document.title=t.meta.title(t)}))}));var Y=n("2f62");r["default"].use(Y["a"]);var G={setMobile:function(t,e){t.mobile=e},setMobileMenu:function(t,e){t.mobileMenu=e},setUser:function(t,e){t.user=e,Object(i["c"])(e.access_token),Object(i["d"])(e.refresh_token)}},K=new Y["a"].Store({state:{mobile:!1,mobileMenu:!1,user:{access_token:"",refresh_token:"",logged_in:!1,email:"",settings:{}}},mutations:G}),W=n("a65d"),Q=n.n(W),V=n("bc3a"),X=n.n(V),Z=n("a7fe"),tt=n.n(Z),et=n("5887"),nt=n.n(et);r["default"].use(nt.a),r["default"].use(tt.a,X.a),r["default"].use(Q.a,{action:{text:"Close",onClick:function(t,e){e.goAway(0)}},position:"bottom-right",duration:5e3,keepOnHover:!0}),r["default"].config.productionTip=!1,new r["default"]({router:F,store:K,render:function(t){return t(E)}}).$mount("#app")},"5c0b":function(t,e,n){"use strict";var r=n("9c0c"),o=n.n(r);o.a},"66ca":function(t,e,n){"use strict";n.d(e,"a",(function(){return s})),n.d(e,"b",(function(){return a})),n.d(e,"c",(function(){return i})),n.d(e,"d",(function(){return c}));var r={access_token:"access_token",refresh_token:"refresh_token"},o=window.localStorage,s=function(){return JSON.parse(o.getItem(r.access_token))},a=function(){return JSON.parse(o.getItem(r.refresh_token))},i=function(t){o.setItem(r.access_token,JSON.stringify(t))},c=function(t){o.setItem(r.refresh_token,JSON.stringify(t))}},"99ec":function(t,e,n){},"9c0c":function(t,e,n){},a3f9:function(t,e,n){"use strict";var r=n("13b9"),o=n.n(r);o.a},af63:function(t,e,n){"use strict";var r=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"main-panel"},[this.$slots.header?n("div",{staticClass:"main-panel-header"},[t._t("header")],2):t._e(),n("div",{staticClass:"main-panel-content"},[t._t("content")],2)])},o=[],s={name:"main-panel",props:{}},a=s,i=(n("4733"),n("2877")),c=Object(i["a"])(a,r,o,!1,null,null,null);e["a"]=c.exports},b1fe:function(t,e,n){},c860:function(t,e,n){"use strict";var r=function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("button",{class:["button",t.typeClass]},[t.isIcon?[n("i",{staticClass:"material-icons"},[t._v(t._s(t.text))])]:[t._v(t._s(t.text))]],2)},o=[],s={name:"v-button",props:{hierachyLevel:{type:String,default:"primary",required:!0},text:{type:String,default:"",required:!0},isIcon:{type:Boolean,default:!1,required:!1}},computed:{typeClass:function(){return"button--".concat(this.hierachyLevel)}}},a=s,i=(n("cbe2"),n("2877")),c=Object(i["a"])(a,r,o,!1,null,null,null);e["a"]=c.exports},cbe2:function(t,e,n){"use strict";var r=n("99ec"),o=n.n(r);o.a},ecf6:function(t,e,n){"use strict";n("99af"),n("d3b7");var r=n("bc3a"),o=n.n(r),s="http://127.0.0.1:5000",a=o.a.create({baseURL:"".concat(s,"/api/")});n.d(e,"d",(function(){return i})),n.d(e,"j",(function(){return c})),n.d(e,"g",(function(){return u})),n.d(e,"l",(function(){return l})),n.d(e,"a",(function(){return d})),n.d(e,"h",(function(){return f})),n.d(e,"i",(function(){return h})),n.d(e,"k",(function(){return g})),n.d(e,"e",(function(){return m})),n.d(e,"c",(function(){return p})),n.d(e,"b",(function(){return b})),n.d(e,"f",(function(){return v}));var i=function(t){var e=new Promise((function(e,n){a.get("account",{headers:{Authorization:"Bearer "+t}}).then((function(t){e(t)})).catch((function(t){n(t)}))}));return e},c=function(t,e){var n=new Promise((function(n,r){a.patch("account",e,{headers:{Authorization:"Bearer "+t}}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},u=function(t,e){var n=new Promise((function(n,r){a.post("login",{email:t,password:e}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},l=function(t,e){var n=new Promise((function(n,r){a.post("account",{email:t,password:e}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},d=function(t,e){var n=new Promise((function(n,r){a.post("accounts/actions/change-password",{reset_token:t,password:e}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},f=function(t){var e=new Promise((function(e,n){a.post("logout/access",{},{headers:{Authorization:"Bearer "+t}}).then((function(t){e(t)})).catch((function(t){n(t)}))}));return e},h=function(t){var e=new Promise((function(e,n){a.post("logout/refresh",{},{headers:{Authorization:"Bearer "+t}}).then((function(t){e(t)})).catch((function(t){n(t)}))}));return e},g=function(t,e,n){var r=new Promise((function(r,o){a.patch("sensors/".concat(e),n,{headers:{Authorization:"Bearer "+t}}).then((function(t){r(t)})).catch((function(t){o(t)}))}));return r},m=function(t,e,n,r,o){var s=new Promise((function(n,s){a.get("sensors/".concat(e,"/climate-data?range_start=").concat(r,"&range_end=").concat(o),{headers:{Authorization:"Bearer "+t}}).then((function(t){n(t)})).catch((function(t){s(t)}))}));return s},p=function(t,e){var n=new Promise((function(n,r){a.delete("sensors/".concat(e),{headers:{Authorization:"Bearer "+t}}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},b=function(t,e){var n=new Promise((function(n,r){a.delete("sensors/".concat(e,"/climate-data"),{headers:{Authorization:"Bearer "+t}}).then((function(t){n(t)})).catch((function(t){r(t)}))}));return n},v=function(t){var e=new Promise((function(e,n){a.get("sensors",{headers:{Authorization:"Bearer "+t}}).then((function(t){e(t)})).catch((function(t){n(t)}))}));return e}}});
//# sourceMappingURL=app.eb7140f7.js.map