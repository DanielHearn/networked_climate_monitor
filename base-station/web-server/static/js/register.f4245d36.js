(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["register"],{"04de":function(t,e,s){"use strict";var r=s("aeaf"),a=s.n(r);a.a},"73cf":function(t,e,s){"use strict";s.r(e);var r=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("main-panel",[s("template",{slot:"content"},[s("div",{staticClass:"content-column"},[s("h1",{staticClass:"title"},[t._v("Climate Monitor")]),s("h2",{staticClass:"heading"},[t._v("Account Registration")]),t.$store.state.user.email?[s("p",{staticClass:"text bold register-success"},[t._v(" The account has been successfully registered ")]),s("p",{staticClass:"text"},[t._v(" Complete the setup instructions to start using the climate monitor system. ")]),s("ol",{staticClass:"ordered-list"},[s("li",[s("p",{staticClass:"text"},[t._v(" Save the reset token below into a safe location such as in a text file or on paper. This token will be required to change or reset the account's password. ")]),s("span",{staticClass:"text bold"},[t._v("Reset token: "+t._s(t.$store.state.user.reset_token))])]),s("li",[s("p",{staticClass:"text"},[t._v(" Connect each sensor node to their battery. ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" Plugin each sensor node's micro usb port to a USB port with a USB cable to charge the node's battery until the battery level on the dashboard is \"High\". ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" Once the sensor nodes have been fully charged place the nodes in the areas that will be monitored. ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" If the sensor nodes or climate data aren't appearing in the dashboard move the nodes closer to the base station so that they can communicate. ")])])]),s("router-link",{staticClass:"link",attrs:{to:"/dashboard"}},[t._v("Completed the setup? Click here to view the dashboard")])]:[s("p",{staticClass:"text"},[t._v(" Enter an email and password to register the account. ")]),s("register-form"),s("router-link",{staticClass:"link",attrs:{to:"/login"}},[t._v("Already registered? Click here to login")])]],2)])],2)},a=[],o=s("af63"),i=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("form",{staticClass:"form",attrs:{id:"register",method:"post"},on:{submit:t.checkRegister}},[s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Email")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.email,expression:"email"}],staticClass:"input--text",attrs:{type:"email",name:"email",placeholder:"Email",tabindex:"0",autocomplete:"email"},domProps:{value:t.email},on:{input:function(e){e.target.composing||(t.email=e.target.value)}}})]),s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Password")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.password,expression:"password"}],staticClass:"input--text",attrs:{type:"password",name:"password",placeholder:"Password",tabindex:"0",autocomplete:"new-password"},domProps:{value:t.password},on:{input:function(e){e.target.composing||(t.password=e.target.value)}}})]),s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Confirm Password")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.confirmPassword,expression:"confirmPassword"}],staticClass:"input--text",attrs:{type:"password",name:"confirmPassword",placeholder:"Confirm password",tabindex:"0",autocomplete:"new-password"},domProps:{value:t.confirmPassword},on:{input:function(e){e.target.composing||(t.confirmPassword=e.target.value)}}})]),t.errors.length?s("error-list",{attrs:{title:"Errors:",errors:t.errors}}):t._e(),s("v-button",{attrs:{hierachyLevel:"primary",text:"Register",type:"submit",tabindex:"0"}})],1)},n=[],c=(s("c975"),s("ac1f"),s("5319"),s("93da")),l=s("ecf6"),d=s("6d06"),u=s("66ca"),p=s("c860"),h={name:"register-form",components:{ErrorList:c["a"],vButton:p["a"]},data:function(){return{email:"",password:"",confirmPassword:"",errors:[]}},methods:{checkRegister:function(t){t.preventDefault(),this.errors=[];var e=this.email,s=this.password,r=this.confirmPassword,a=!0;""===e?(this.errors.push("Enter a email."),a=!1):-1===e.indexOf("@")&&(this.errors.push("Enter a valid email."),a=!1),""===s?(this.errors.push("Enter a password."),a=!1):(s.length<8||s.length>40)&&(this.errors.push("Password must be between 8 and 40 characters long."),a=!1),s!==r&&(this.errors.push("Password and confirm password be identical."),a=!1),a&&this.register(e,s)},register:function(t,e){var s=this;this.$toasted.show("Sending register request"),Object(l["l"])(t,e).then((function(e){var r=e.data;if(r.status&&r.access_token&&r.refresh_token){var a=s.$store.state.user;a.logged_in=!0,a.access_token=r.access_token,a.refresh_token=r.refresh_token,a.email=t,Object(u["c"])(r.access_token),Object(u["d"])(r.refresh_token),Object(l["d"])(r.access_token).then((function(t){var e=t.data;if(e.status&&e.account){var r=s.$store.state.user;r.settings=JSON.parse(e.account.settings.replace(/'/g,'"')),s.$store.commit("setUser",r)}})).catch((function(t){console.warn(t)})),s.$store.commit("setUser",a),s.$toasted.show("Registered")}})).catch((function(t){t.response&&t.response.data&&t.response.data.errors?s.errors=Object(d["d"])(t.response.data.errors):s.errors.push(t)}))}}},m=h,f=(s("04de"),s("2877")),v=Object(f["a"])(m,i,n,!1,null,null,null),g=v.exports,w={name:"register",components:{MainPanel:o["a"],RegisterForm:g}},_=w,b=Object(f["a"])(_,r,a,!1,null,null,null);e["default"]=b.exports},"7ac5":function(t,e,s){"use strict";var r=s("e56d"),a=s.n(r);a.a},"93da":function(t,e,s){"use strict";var r=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("div",{staticClass:"error-list"},[t.title?s("p",{staticClass:"title"},[t._v(t._s(t.title))]):t._e(),s("ul",{staticClass:"errors"},t._l(t.errors,(function(e,r){return s("li",{key:r,staticClass:"error"},[s("p",[t._v(t._s(e))])])})),0)])},a=[],o={name:"error-list",props:{title:{type:String,default:"",required:!1},errors:{type:Array,default:[],required:!0}}},i=o,n=(s("7ac5"),s("2877")),c=Object(n["a"])(i,r,a,!1,null,null,null);e["a"]=c.exports},a640:function(t,e,s){"use strict";var r=s("d039");t.exports=function(t,e){var s=[][t];return!!s&&r((function(){s.call(null,e||function(){throw 1},1)}))}},aeaf:function(t,e,s){},c975:function(t,e,s){"use strict";var r=s("23e7"),a=s("4d64").indexOf,o=s("a640"),i=s("ae40"),n=[].indexOf,c=!!n&&1/[1].indexOf(1,-0)<0,l=o("indexOf"),d=i("indexOf",{ACCESSORS:!0,1:0});r({target:"Array",proto:!0,forced:c||!l||!d},{indexOf:function(t){return c?n.apply(this,arguments)||0:a(this,t,arguments.length>1?arguments[1]:void 0)}})},e56d:function(t,e,s){}}]);
//# sourceMappingURL=register.f4245d36.js.map