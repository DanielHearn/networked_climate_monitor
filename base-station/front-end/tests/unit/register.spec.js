import Vuex from 'vuex'
import { shallowMount } from '@vue/test-utils'
import Register from './../../src/views/Register.vue'
import { storeConfig } from './../../src/store/index.js'

describe('Register.vue', () => {
  let wrapper

  beforeEach(() => {
    const store = new Vuex.Store(storeConfig)
    wrapper = shallowMount(Register, {
      stubs: ['router-link'],
      store
    })
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
