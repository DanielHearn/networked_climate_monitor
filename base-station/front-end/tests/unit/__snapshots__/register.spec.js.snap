// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register.vue has the expected html structure 1`] = `
<main-panel-stub>
  <template>
    <div
      class="content-column"
    >
      <h1
        class="title"
      >
        Climate Monitor
      </h1>
       
      <h2
        class="heading"
      >
        Account Registration
      </h2>
       
      <p
        class="text"
      >
        
          Enter an email and password to register the account.
        
      </p>
       
      <register-form-stub />
       
      <router-link-stub
        class="link"
        to="/login"
      >
        Already registered? Click here to login
      </router-link-stub>
    </div>
  </template>
</main-panel-stub>
`;
